tools/dm/skip-replace-sqls.md

```diff
@@ -0,0 +1,530 @@
+
+However, when you try to avoid breaking the replication by actively handling unsupported SQL statements, you cannot know in advance the position of the binlog event, so you need another approach to match the subsequent binlog events.
+
+In DM, two modes of matching the binlog event are supported (you can only choose one mode from below):
+1. binlog position: the position information of the binlog event 
+    - The binlog position is given by `--binlog-pos` in the command, and the format is `binlog-filename:binlog-pos`, for example, `mysql-bin|000001.000003:3270`.
+    - The format of the binlog filename in DM is not completely consistent with that in the upstream MySQL.
+    - When the replication error occurs, the position can be directly obtained from `failedBinlogPosition` returned by `query-error`.
+2. DDL pattern: the regular expression (only for the DDL statement) matching mode
+    - The DDL pattern is given by `--sql-pattern` in the command, for example, to match ``` ALTER TABLE `db2`.`tbl2` DROP COLUMN `c2` ```, the corresponding regular expression should be ``` ~(?i)ALTER\s+TABLE\s+`db2`.`tbl2`\s+DROP\s+COLUMN\s+`c2` ```.
+    - The regular expression must be prefixed with `~` and cannot contain any common space (you can replace the space with `\s` or `\s+` in the string).
+
+In the scenario of merging and replicating data from sharded tables, if you need DM to automatically select a DDL lock owner to execute the skip/replace operation, then you must use the DDL pattern matching mode because the binlog positions corresponding to the DDL statements on different DM-workers have no logical connection and are hard to confirm.
+
+**Notes:**
+
+- You can only register one operator (specified by `--binlog-pos`) for one binlog event. The previous one can be overwritten by the newly registered operator.
+- Do not specify an operator for one binlog event by using `--binlog-pos` and `--sql-pattern` at the same time.
+- After the operator successfully matches with the binlog event (not after the execution succeeds), it will be deleted. If you need to match again (using `--sql-pattern`) later, you have to register a new operator.
+
+### Support Scenarios
+
+- During the replication, if the DDL statement unsupported by TiDB is executed in the upstream and replicated to the downstream, the replication task will get interrupted.
+    - If it is acceptable that this DDL statement is skipped in the downstream TiDB, then you can use `sql-skip` to resume the replication.
+    - If it is acceptable that this DDL statement is replaced with other DDL statements, then you can use `sql-replace` to resume the replication.
+- During the replication, if you know in advance that an unsupported SQL statement is going to be replicated, you can handle it beforehand to avoid breaking the replication.
```

Comment:
```suggestion
- Scenario 2: During the replication, if you know in advance that an unsupported SQL statement is going to be replicated, you can handle it beforehand to avoid breaking the replication.
```